src/chunker/layout.js

```diff
@@ -467,233 +530,572 @@ class Layout {
+				continue;
+			}
 
-					let tableRow;
-					if (node.nodeName === "TR") {
-						tableRow = node;
-					} else {
-						tableRow = parentOf(node, "TR", rendered);
-					}
-					if (tableRow) {
-						// honor break-inside="avoid" in parent tbody/thead
-						let container = tableRow.parentElement;
-						if (["TBODY", "THEAD"].includes(container.nodeName)) {
-							let styles = window.getComputedStyle(container);
-							if (styles.getPropertyValue("break-inside") === "avoid") prev = container;
-						}
+			if (left < bLeft || right > bRight || top < bTop || bottom > bBottom) {
+				return child;
+			}
+		}
+	}
 
-						// Check if the node is inside a row with a rowspan
-						const table = parentOf(tableRow, "TABLE", rendered);
-						const rowspan = table.querySelector("[colspan]");
-						if (table && rowspan) {
-							let columnCount = 0;
-							for (const cell of Array.from(table.rows[0].cells)) {
-								columnCount += parseInt(cell.getAttribute("colspan") || "1");
-							}
-							if (tableRow.cells.length !== columnCount) {
-								let previousRow = tableRow.previousElementSibling;
-								let previousRowColumnCount;
-								while (previousRow !== null) {
-									previousRowColumnCount = 0;
-									for (const cell of Array.from(previousRow.cells)) {
-										previousRowColumnCount += parseInt(cell.getAttribute("colspan") || "1");
-									}
-									if (previousRowColumnCount === columnCount) {
-										break;
-									}
-									previousRow = previousRow.previousElementSibling;
-								}
-								if (previousRowColumnCount === columnCount) {
-									prev = previousRow;
-								}
+	startOfOverflow(node, bounds) {
+		let childNode, done = false;
+		let prev;
+		let anyOverflowFound = false;
+
+		do {
+			prev = node;
+			do {
+				childNode = this.firstOverflowingChild(node, bounds);
+				if (childNode) {
+					anyOverflowFound = true;
+				} else {
+					// The overflow isn't caused by children. It could be caused by:
+					// * a sibling div / td / element with height that stretches this
+					//   element
+					// * margin / padding on this element
+					// In the former case, we want to ignore this node and take the
+					// sibling. In the later case, we want to move this node.
+					let intrinsicBottom = 0, intrinsicRight = 0;
+					if (isElement(node)) {
+						let styles = window.getComputedStyle(node);
+
+						if (node.childNodes.length) {
+							let lastChild = node.childNodes[node.childNodes.length - 1];
+							let childBounds = getBoundingClientRect(lastChild);
+							intrinsicRight = childBounds.right;
+							intrinsicBottom = childBounds.bottom;
+						} else {
+							// Has no children so should have no height, all other things
+							// being equal.
+							let childBounds = getBoundingClientRect(node);
+							intrinsicRight = getBoundingClientRect(node).right;
+							intrinsicBottom = getBoundingClientRect(node).top;
+							let intrinsicLeft = getBoundingClientRect(node).x;
```

Comment:
Can we use `childBounds` instead of `getBoundingClientRect` again for the next few lines?

```suggestion
							let childBounds = getBoundingClientRect(node);
							intrinsicRight = childBounds.right;
							intrinsicBottom = childBounds.top;
							let intrinsicLeft = childBounds.x;
```

Reply:
Will check.

Reply:
When I actually read the code, there's nothing to "check". The answer should have been "yes, oops!". Done now!